lib/teleterm/gateway/gateway.go

```diff
@@ -207,14 +220,32 @@ func (g *Gateway) CLICommand() (string, error) {
 	return cliCommand, nil
 }
 
-// Gateway describes local proxy that creates a gateway to the remote Teleport resource.
+// ReloadCert loads the key pair from cfg.CertPath & cfg.KeyPath and updates the cert of the running
+// local proxy. This is typically done after the cert is reissued and saved to disk.
 //
-// Gateway is not safe for concurrent use in itself. However, all access to gateways is gated by
-// daemon.Service which obtains a lock for any operation pertaining to gateways.
-//
-// In the future if Gateway becomes more complex it might be worthwhile to add an RWMutex to it.
+// In the future, we're probably going to make this method accept the cert as an arg rather than
+// reading from disk.
+func (g *Gateway) ReloadCert() error {
+	g.mu.Lock()
```

Comment:
Now that I'm looking at this, I'm not sure if this mutex here is needed. `alpn.LocalProxy` processes connections serially, so there's no risk of two separate goroutines trying to set the cert.

Reply:
While reviewing this code I realized I needed a mutex in the local proxy still.

See: #18278

The window for a data race is extremely small, but we do call GetCerts to copy the certs into the TLS config when we dial upstream.

I originally had a mutex in there, but then we made the middleware a blocking call and I thought it was safe to remove.

That being said, once that PR merges I don't think you need a mutex here as the local proxy will guard its own certs with a mutex.